## Data Analysis Using SQL and Pandas on **Chinkook Database**

### Importing Libraries

In [40]:
import pandas as pd
import sqlite3

### Connecting To my Sqlite file


In [41]:
conn = sqlite3.connect('Chinook_Sqlite.sqlite')
cursor = conn.cursor()

### Sanity Checking That All Tables are Loaded

In [42]:
pd.read_sql_query("SELECT name FROM sqlite_master WHERE type = 'table';",conn)

,name
0,Album
1,Artist
2,Customer
3,Employee
4,Genre
5,Invoice
6,InvoiceLine
7,MediaType
8,Playlist
9,PlaylistTrack


### Top **5** Genres With the Most Revenue

In [43]:
pd.read_sql_query("SELECT g.GenreId,g.Name,SUM(il.UnitPrice*il.Quantity) AS total_revenue FROM Genre g JOIN Track t ON g.GenreId = t.GenreId JOIN InvoiceLine il ON il.TrackId = t.TrackId GROUP BY g.GenreID,g.Name ORDER BY total_revenue DESC LIMIT 5",conn)

,GenreId,Name,total_revenue
0,1,Rock,826.65
1,7,Latin,382.14
2,3,Metal,261.36
3,4,Alternative & Punk,241.56
4,19,TV Shows,93.53


##### This Table shows us that **Rock** has the highest revenue of **826.65 dollars**.
##### On the other hand, **TV Shows** has the lowest revenue of **93.53**